### PR TITLE
Fix a few bugs and code smells, but mostly convert to autotools

### PR DESCRIPTION
**Pull request by jikamens**
The best way to review this PR is probably to review the commits one by one rather than reviewing them all at once, since each commit stands alone.

I have confirmed that the code in this PR builds and installs successfully on Debian Linux, FreeBSD, and macOS.

### CODE REVIEW

**Comment by fangfufu** [comment]
Thanks, I can see you put in a lot of efforts, and your code quality is really good. I have added you as a collaborator. You can now push into this project directly. :)

**Comment by jikamens** [comment]
@fangfufu I appreciate your confidence but I will still probably ask you to review any future changes that I want to commit, unless they are quite minor. Two pairs of eyes are better than one!

**Comment by fangfufu** [comment]
@jikamens , I don't give collaborator access out to everyone - the only other person I have given it to is my Debian package maintainer.

File: src/link.c [hunk 1]
```diff
@@ -724,16 +739,28 @@ LinkTable *LinkTable_disk_open(const char *dirn)
 
     LinkTable *linktbl = CALLOC(1, sizeof(LinkTable));
 
-    fread(&linktbl->num, sizeof(int), 1, fp);
+    if (sizeof(int) != fread(&linktbl->num, sizeof(int), 1, fp)) {
```
Comment (fangfufu) [inline src/link.c:742]: Yikes, 
> On  success, fread() and fwrite() return the number of items read or written.  This number equals the number of bytes transferred only when size is 1.  If an error occurs, or the end of the file is reached, the return value is a short item count (or zero).

Comment (fangfufu) [reply src/link.c:742]: This makes me wonder if the cache system has been broken on the master branch since last September...

Comment (fangfufu) [reply src/link.c:742]: The point is that ``sizeof(int)`` is 4, but we are only expecting ``1`` to be returned here, as we are reading 1 int.

File: src/cache.c [hunk 2]
```diff
@@ -163,15 +163,11 @@ static int Meta_read(Cache *cf)
...
-    fread(&cf->segbc, sizeof(long), 1, fp);
-
-    /*
-     * Error checking for fread
-     */
-    if (ferror(fp)) {
+    if (sizeof(long) != fread(&cf->time, sizeof(long), 1, fp) ||
+        sizeof(off_t) != fread(&cf->content_length, sizeof(off_t), 1, fp) ||
+        sizeof(int) != fread(&cf->blksz, sizeof(int), 1, fp) ||
+        sizeof(long) != fread(&cf->segbc, sizeof(long), 1, fp) ||
```
Comment (fangfufu) [inline src/cache.c:169]: These are all wrong too. I have fixed them here:
https://github.com/fangfufu/httpdirfs/commit/28293b5ccdc4f65e5721ba2fe978f1cff066d0d7

**Comment by jikamens** [comment]
Thanks for catching those errors! Sorry about that.

**Pull request merged**